src/main/java/edu/greenblitz/tobyDetermined/commands/telescopicArm/moveByNodes/StraightLineMotion.java

```diff
@@ -0,0 +1,146 @@
+    private static final double MAX_EXTENDER_VELOCITY = 1.5; //Meters Per Second
+    private static final double MAX_ANGULAR_VELOCITY = 3;//Radians Per Second
+    private static final double TO_BELLY_VELOCITY = 0.2;
+    private static final double ONLY_ELBOW_VELOCITY = 1;
+
+    public StraightLineMotion(SystemsState start, SystemsState end) {
+        extender = Extender.getInstance();
+        elbow = Elbow.getInstance();
+        require(elbow);
+        require(extender);
+        combinedVelocity = 0;
+        this.startState = start;
+        this.endState = end;
+        startingNode = (NodeArm) getNode(start);
+        endNode = (NodeArm) getNode(end);
+    }
+
+    @Override
+    public void initialize() {
+        if (extender.getLength() + endNode.getExtendPosition() <= DISTANCE_NOT_STRAIGHT_LINE)
+            combinedVelocity = ONLY_ELBOW_VELOCITY;
+
+        else if (endState.equals(SystemsState.ARM_GROUND))
+            combinedVelocity = TO_BELLY_VELOCITY;
+
+        else {
+            double length1 = startingNode.getExtendPosition() + STARTING_LENGTH;
+            double length2 = endNode.getExtendPosition() + STARTING_LENGTH;
+            double distance = GBMath.distance(
+                    GBMath.convertToCartesian(length1, startingNode.getAnglePosition()),
+                    GBMath.convertToCartesian(length2, endNode.getAnglePosition())
+            );
+            combinedVelocity = STATIC_COMBINED_VELOCITY * GBMath.sigmoid(distance, 1, 5.9, 0.3);//numbers that work well
+        }
+
+        Logger.getInstance().recordOutput("Arm/TargetPose3D/endNode", ArmSimulation.getArmPosition(endNode.getExtendPosition(), (endNode.getAnglePosition())));
+    }
+
+    public static double getRatioBetweenAngleAndLength(double sideA, double sideB, double gamma) {
+        double sideC = GBMath.lawOfCosines(sideA, sideB, gamma);
+        double height = sideB * Math.sin(gamma);
+        double adjacent = Math.sqrt(sideC * sideC - height * height);
+        return height / adjacent;
+    }
+
+    public double calculateExtenderVelocity(double ratio) {
+        double signOfExtender = Math.signum(endNode.getExtendPosition() - extender.getLength());
+        double extenderVelocity = Math.sqrt(combinedVelocity * combinedVelocity / (ratio * ratio + 1));
+        double wantedVelocity = signOfExtender * extenderVelocity;
+
+        wantedVelocity = limit(wantedVelocity, MAX_EXTENDER_VELOCITY);
+
+        return wantedVelocity;
+    }
+
+    public double calculateAngularVelocity(double startVelocity) {
+        double signOfAngle = Math.signum(endNode.getAnglePosition() - elbow.getAngleRadians());
+        double magnitudeOfVelocity = startVelocity / (extender.getLength() + STARTING_LENGTH);
+
+        magnitudeOfVelocity = limit(magnitudeOfVelocity, MAX_ANGULAR_VELOCITY);
+
+        return signOfAngle * Math.abs(magnitudeOfVelocity);
+    }
+
+    public double limit(double v, double maxMagnitude) {
```

Comment:
What is v?

Reply:
value can be anything you want to limit, I changed name to value